File: C++/ShuffleAnArray.cpp
class Solution {
public:
    vector<int> nums;
    Solution(vector<int>& nums) {
        this->nums=nums;
    }
    
    /** Resets the array to its original configuration and return it. */
    vector<int> reset() {
        return nums;
    }
    
    /** Returns a random shuffling of the array. */
    vector<int> shuffle() {
        vector<int> nw(nums);
        for(int i=0;i<nums.size();i++){
            int pos=rand()%(nums.size()-i);
            swap(nw[pos+i],nw[i]);
        }
        return nw;
    }
};

/**
 * Your Solution object will be instantiated and called as such:
 * Solution* obj = new Solution(nums);
 * vector<int> param_1 = obj->reset();
 * vector<int> param_2 = obj->shuffle();
 */

File: Python/ShuffleAnArray.py
#https://leetcode.com/problems/shuffle-an-array/
class Solution:
    def __init__(self, nums):
        self.arr = nums
        self.original = list(nums)

    def reset(self):
        self.arr = self.original
        self.original = list(self.original)
        return self.arr

    def shuffle(self):
        aux = list(self.arr)

        for idx in range(len(self.arr)):
            remove_idx = random.randrange(len(aux))
            self.arr[idx] = aux.pop(remove_idx)

        return self.arr
